### ISSUE TITLE
Break should not skip finally clauses

### ISSUE THREAD

**Issue by cpcallen** [NeilFraser/JS-Interpreter#93]
In this code:

    var a = 0
    while(true){
        try {
            break
        }
        finally {
            a++
        }
    }
    alert(a)

The final value of a should be 1, not 0.

**Comment by NeilFraser** [comment]
Same applies to `continue`:
```
var a = 0;
for (var i = 0; i < 10; i++) {
  try {
    continue;
  }
  finally {
    a++;
  }
}
alert(a);
```

**Comment by NeilFraser** [comment]
For reference, this one is also fun.  It should alert 10:
```
for (var i = 0; i < 10; i++) {
  try {
    break;
  } finally {
    continue;
  }
}
alert(i);
```

**Comment by NeilFraser** [comment]
Holy Zarquon's singing fish...
```
function foo() {
  for (var i = 0; i < 10; i++) {
    try {
      return 42;
    } finally {
      continue;
    }
  }
  return i;
}
alert(foo());
```
This returns 10.

**Comment by cpcallen** [comment]
The frowny-face reaction isn't really strong enough.  Here's my true feeling: 😱

**Comment by NeilFraser** [comment]
Fixed in #120.
